digest.go

```diff
@@ -224,6 +232,12 @@ func (t *Transport) newCredentials(req *http.Request, c *challenge) *credentials
 		method:     req.Method,
 		password:   t.Password,
 	}
+	if c.Algorithm == "MD5" {
```

Comment:
we should probably error here earlier on if we're going to put the struct in a bad state